sdk/loadtestservice/load-testing-rest/src/getFileValidationPoller.ts

```diff
@@ -0,0 +1,165 @@
+import { TestGetFile200Response, TestUploadFile201Response } from "./responses";
+import { isUnexpected } from "./isUnexpected";
+import { sleep } from "./util/LROUtil";
+
+/**
+ * Uploads a file and creates a poller to poll for validation.
+ * @param client - The Load Testing client.
+ * @param options - The operation options.
+ * @returns A poller which can be called to poll until completion of the job.
+ */
+export async function getFileValidationPoller(
+  client: AzureLoadTestingClient,
+  fileUploadResult: TestUploadFile201Response,
+  polledOperationOptions: PolledOperationOptions = {}
+): Promise<FileUploadAndValidatePoller> {
+  // get filename and testid from initial response
+  const fileName = fileUploadResult.body.fileName;
+  const requestUrl = fileUploadResult.request.url;
+  const testId = requestUrl.substring(
+    requestUrl.indexOf("tests/") + 6,
+    requestUrl.lastIndexOf("/files")
+  );
+  type Handler = (state: OperationState<TestGetFile200Response>) => void;
+
+  const state: OperationState<TestGetFile200Response> = {
+    status: "notStarted",
+  };
+
+  const progressCallbacks = new Map<symbol, Handler>();
+  const processProgressCallbacks = async (): Promise<void> =>
+    progressCallbacks.forEach((h) => h(state));
+  let resultPromise: Promise<TestGetFile200Response> | undefined;
+  let cancelJob: (() => void) | undefined;
+  const abortController = new AbortController();
+  const currentPollIntervalInMs = polledOperationOptions.updateIntervalInMs ?? 2000;
+
+  const poller: SimplePollerLike<OperationState<TestGetFile200Response>, TestGetFile200Response> = {
+    async poll(options?: { abortSignal?: AbortSignalLike }): Promise<void> {
+      if (options?.abortSignal?.aborted) {
+        state.error = new Error("The operation was aborted.");
+        return;
+      }
+
+      if (fileName) {
+        const fileValidationResponse = await client
+          .path("/tests/{testId}/files/{fileName}", testId, fileName)
+          .get();
+        if (isUnexpected(fileValidationResponse)) {
+          state.status = "failed";
+          state.error = new Error(fileValidationResponse.body.error.message);
+          return;
+        }
+
+        switch (fileValidationResponse.body.validationStatus) {
+          case "NOT_VALIDATED": {
+            if (fileValidationResponse.body.fileType === "JMX_FILE") {
+              state.status = "running";
+            } else {
+              state.status = "succeeded";
+            }
+            break;
+          }
+          case "VALIDATION_INITIATED": {
+            state.status = "running";
+            break;
+          }
+          case "VALIDATION_SUCCESS":
+          case "VALIDATION_NOT_REQUIRED": {
+            state.status = "succeeded";
+            break;
+          }
+          case "VALIDATION_FAILURE": {
+            state.status = "failed";
+            state.error = new Error(fileValidationResponse.body.validationStatus);
```

Comment:
I wonder if there is something more informative that can be used as the error message?

Reply:
`fileValidationResponse.body.error.message`?

Reply:
Updated error message to have fileValidationResponse.body.validationFailureDetails which contains details of failure.